OnIncomingEvent/events.ts

```diff
@@ -0,0 +1,42 @@
+import * as t from "io-ts";
+import { withDefault } from "@pagopa/ts-commons/lib/types";
+import { NonEmptyString } from "@pagopa/ts-commons/lib/strings";
+
+const empty = withDefault(t.any, {} as const);
+
+interface IBaseEvent<N extends string, P> {
+  readonly name: N;
+  readonly payload: P;
+}
+
+// helper to define a codec for an event
+const event = <N extends string, P>(
+  name: N,
+  payload: t.Type<P> = empty
+): t.Type<IBaseEvent<N, P>> => t.interface({ name: t.literal(name), payload });
+
+type PingOneEvent = t.TypeOf<typeof PingOneEvent>;
+const PingOneEvent = event(
+  "ping" as const,
+  t.interface({ name: NonEmptyString })
+);
+
+type PingAllEvent = t.TypeOf<typeof PingAllEvent>;
+const PingAllEvent = event("ping:all" as const);
+
+export type PublicEvent = t.TypeOf<typeof PublicEvent>;
+export const PublicEvent = PingOneEvent; // this will be a union of all public events
+
+export type NonPublicEvent = t.TypeOf<typeof NonPublicEvent>;
+export const NonPublicEvent = PingAllEvent; // this will be a union of all non-public events
+
+export type IncomingEvent = t.TypeOf<typeof IncomingEvent>;
+export const IncomingEvent = t.union([PublicEvent, NonPublicEvent]);
+
+// Map every incoming event with the eventual set of required attributes
+export const publicEventsRequiredAttributes: Record<
+  PublicEvent["name"],
+  t.Type<Record<string, unknown>>
+> = {
+  ping: empty
+};
```

Comment:
Added new test in [0bbce58](https://github.com/pagopa/io-functions-public-event-dispatcher/pull/6/commits/0bbce587d5083a3a9affb270a9435107e4f78341) and [2a27a00](https://github.com/pagopa/io-functions-public-event-dispatcher/pull/6/commits/2a27a00e3f29edc1ae584a89ef2d22b22267bc6d)

Reply:
if I get it right, I'm expecting `ping` to be a PublicEvent to be notified only to a specific webhook (the one with the corresponding `name` attribute). 
This is not happening in case a webhook has been defined with no `name` atribute, it will be notified too.

Reply:
Deefining `publicEventsRequiredAttributes` this way will fix the test:

```typescript
export const publicEventsRequiredAttributes: Record<
  PublicEvent["name"],
  t.Mixed
> = {
  ping: t.interface({ name: NonEmptyString })
};
```

Reply:
Well spotted, thank you! Fixed in https://github.com/pagopa/io-functions-public-event-dispatcher/pull/6/commits/6bc92685c3fcce7374b6579ae5af1fd4f0b292d2